### DISCUSSION TITLE
How to install extensions?

### DISCUSSION THREAD

**Discussion by YouveGotMeowxy** [podman-desktop/podman-desktop#169] [Q&A] [answered]
Hi

I dl'd the latest PD and want to install the docker extension now, but cant find any documentation, or obvious way of doing so via the GUI?

How can I do it?

TIA

**Comment by benoitf** [comment]
Hello @YouveGotMeowxy for now there is no support of Docker Desktop extension.

There is an independent plug-in mechanism allowing for now low-level/non-UI contributions.

What was the extension you were interested in ?

**Comment by YouveGotMeowxy** [comment]
I dunno, lol.

The website just shows that there's a Docker extension available for it:

![image](https://user-images.githubusercontent.com/39491297/171886131-54fd7b73-fd74-4136-a8ca-bd8ceb74abae.png)

And there's [this](https://github.com/containers/podman-desktop/tree/main/extensions/docker).

**Comment by benoitf** [comment]
ok got it. 

The docker extension is not embedded in the assembly by default
Source code is there https://github.com/containers/podman-desktop/tree/main/extensions/docker

you can go to this folder and run `yarn build`
it produces a `docker.cdix` file that can be dropped into podman desktop folder that is scanned at runtime


so basically:

1. create `$HOME/.local/share/podman-desktop/plugins/` folder
2. build .cdix (or grab it from this issue I've attached it (but it's inside a zip)
          [docker.zip](https://github.com/containers/podman-desktop/files/8833438/docker.zip)
 3. cp docker.cdix ~/.local/share/podman-desktop/plugins/

You should see Docker Desktop containers/images/etc and the new extension there

**Reply by YouveGotMeowxy** [reply]
@benoitf thank you for saving me from having to compile that myself. :)

I followed your instructions, but PD just doesnt seem to want to find the plugin.

![image](https://user-images.githubusercontent.com/39491297/171982125-b6919a85-41b0-4be7-b1ba-71e92430bb5a.png)


Also, I've noticed there's a folder that gets created by PD: `C:\Users\<user>\.local\share\containers\podman-desktop`

I've tried also moving the plugins folder and the plugin into that folder, under that path, but still the same result.

**Comment by benoitf** [comment]
Hello, only the .cdix file needs to be moved into the folder

**Reply by YouveGotMeowxy** [reply]
> Hello, only the .cdix file needs to be moved into the folder

Just for the sake of testing, I placed a copy of the `.cdix` in every single one of the folders, assuming PD would then be able to find the right one. The structure of `C:\Users\<user>\.local\` now looks like this:

```
+-- share
    +-- containers
        +-- docker.cdix
        +-- podman-desktop
            +-- configuration
            ¦   +-- docker.cdix
            ¦   +-- settings.json
            +-- docker.cdix
            +-- plugins
                +-- docker.cdix
```

And the result is the same; the plugin doesn't get loaded, and it doesn't show in the Preferences -> Extension catalog either.

As a side note, rather than create yet another folder (`.local`, which is Linux) just for PD, there's a built-in `Local` folder in Windows where programs normally place their stuff, at: `C:\Users\<user>\AppData\Local` and can be retrieve via the environment variable `%LocalAppData%`. It might be better to add some OS detection code to PD, and if Windows is detected, place the PD configs there instead, as good programming practice. :)

**Reply by YouveGotMeowxy** [reply]
Even running PD as Admin doesn't help. I thought maybe it might help with finding the hidden `.local` folder, but it doesn't.

**Comment by benoitf** [comment, marked as answer]
Hello @YouveGotMeowxy 

I've created a draft PR to embed docker plug-in directly in the assembly so you won't need to install it later
you can find the binary there: https://github.com/containers/podman-desktop/actions/runs/2459461452

then go to the artifacts and download the windows one

https://user-images.githubusercontent.com/436777/172549939-1c535f88-0694-4219-b8dd-18ad9189ab95.mp4


About the path to be in AppData, you're absolutely right. I will change the location to be more Windows friendly when running on Windows.

**Comment by YouveGotMeowxy** [comment]
@benoitf that did it, TY! :)
